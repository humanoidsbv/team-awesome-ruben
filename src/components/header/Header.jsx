import React from 'react';
import styles from './Header.module.css';

import Logo from '../logo/Logo';
import Navigation from '../navigation/Navigation';
import NavigationToggle from '../navigation-toggle/NavigationToggle';
import User from '../user/User';

/*
    Header ** aanpassen naamval
        Navigatie
        User
        Toggle
        Logo

    2 schermen in Visual Studio Code
    Open progamma's in desktop view zien?
*/

const Header = () => (
  <header className={styles.header}>
    <Logo></Logo>
    <Navigation></Navigation>
    <NavigationToggle></NavigationToggle>
    <User></User>
  </header>
);

export default Header;
