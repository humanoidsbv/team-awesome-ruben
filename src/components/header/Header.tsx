import React, { useState } from 'react';

import Logo from '../logo/';
import Navigation from '../navigation/';
import NavigationToggle from '../navigation-toggle/';
import ProfileButton from '../profile-button/';
import styles from './Header.module.css';

const Header = (): React.ReactElement => {
  const [isMenuVisible, setIsMenuVisible] = useState(false);

  const handleClick = () => setIsMenuVisible(!isMenuVisible);

  return (
    <header
      className={`${styles.header} ${
        isMenuVisible ? styles.headerMobileOpen : ''
      }`}
    >
      <Logo />
      <NavigationToggle
        isMenuVisible={isMenuVisible}
        toggleMenu={handleClick}
      />
      <Navigation isMenuVisible={isMenuVisible} />
      <ProfileButton />
    </header>
  );
};

export default Header;
