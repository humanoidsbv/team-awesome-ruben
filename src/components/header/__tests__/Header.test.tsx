import * as React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

it('matches component output with expected output', () => {
  const header = shallow(<Header />);
  expect(header).toMatchSnapshot();
});
