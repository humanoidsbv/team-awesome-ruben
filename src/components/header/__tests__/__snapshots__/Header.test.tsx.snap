// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches component output with expected output 1`] = `
<header
  className="header "
>
  <Logo />
  <NavigationToggle
    isMenuVisible={false}
    toggleMenu={[Function]}
  />
  <Navigation
    isMenuVisible={false}
  />
  <ProfileButton />
</header>
`;
